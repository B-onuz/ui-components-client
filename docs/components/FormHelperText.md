# FormHelperText (Component)

[Descrição do componente e onde usar]

## Como utilizar

```js
import FormHelperText from 'ui-components/lib/components/FormHelperText'
```

```jsx
<FormHelperText />
```

## Variações

### Variant

```jsx
<FormHelperText variant={'variant'} />
```

Escala no tema: `theme.variantScale`

## Styled-system props

- space https://styled-system.com/table#space
- layout https://styled-system.com/table#layout
