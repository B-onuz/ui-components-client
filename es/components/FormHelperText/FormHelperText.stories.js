'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.FormHelperTextExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _FormHelperText = _interopRequireDefault(require('./FormHelperText'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/FormHelperText/FormHelperText.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'FormHelperText',
  component: _FormHelperText['default'],
}
exports['default'] = _default

var FormHelperTextExample = function FormHelperTextExample() {
  return /*#__PURE__*/ _react['default'].createElement(_FormHelperText['default'], {
    color: 'danger',
    children: 'lorem ipsum dolor sit amet lorem amet ipsum',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 9,
      columnNumber: 44,
    },
  })
}

exports.FormHelperTextExample = FormHelperTextExample
