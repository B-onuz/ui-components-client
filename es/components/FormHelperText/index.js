"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _FormHelperText = _interopRequireDefault(require("./FormHelperText"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _FormHelperText["default"];
exports["default"] = _default;