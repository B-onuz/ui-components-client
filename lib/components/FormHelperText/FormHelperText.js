"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Typography = _interopRequireDefault(require("../Typography"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/FormHelperText/FormHelperText.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var FormHelperText = function FormHelperText(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return children ? /*#__PURE__*/_react["default"].createElement(_Typography["default"], _extends({
    as: 'span',
    fontSize: 'small',
    children: children
  }, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 6,
      columnNumber: 21
    }
  })) : null;
};

var _default = FormHelperText;
exports["default"] = _default;
module.exports = exports.default;