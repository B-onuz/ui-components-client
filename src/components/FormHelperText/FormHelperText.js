import React from 'react'
import styled from 'styled-components'
import Typography from '../Typography'

const FormHelperText = ({ children, ...rest }) => {
  return children ? <Typography as={'span'} fontSize={'small'} children={children} {...rest}></Typography> : null
}

export default FormHelperText
