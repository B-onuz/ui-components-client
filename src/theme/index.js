const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
