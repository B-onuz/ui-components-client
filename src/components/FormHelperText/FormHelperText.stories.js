import React from 'react'
import FormHelperText from './FormHelperText'

export default {
  title: 'FormHelperText',
  component: FormHelperText,
}

export const FormHelperTextExample = () => <FormHelperText color={'danger'} children={'lorem ipsum dolor sit amet lorem amet ipsum'} />
