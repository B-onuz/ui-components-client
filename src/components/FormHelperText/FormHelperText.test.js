import React from 'react'
import FormHelperText from './FormHelperText.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render FormHelperText', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <FormHelperText />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
