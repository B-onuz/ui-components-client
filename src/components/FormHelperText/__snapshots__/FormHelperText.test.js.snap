// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render FormHelperText 1`] = `null`;
